packages/sqlite_async/lib/src/web/database/web_sqlite_database.dart

```diff
@@ -85,9 +86,27 @@ class SqliteDatabaseImpl
   Future<void> _init() async {
     _connection = await openFactory.openConnection(SqliteOpenOptions(
         primaryConnection: true, readOnly: false, mutex: mutex)) as WebDatabase;
-    _connection.updates.forEach((update) {
-      updatesController.add(update);
-    });
+
+    final broadcastUpdates = _connection.broadcastUpdates;
+    if (broadcastUpdates == null) {
+      // We can use updates directly from the database.
+      _connection.updates.forEach((update) {
+        updatesController.add(update);
+      });
+    } else {
+      _connection.updates.forEach((update) {
+        updatesController.add(update);
+
+        // Share local updates with other tabs
+        broadcastUpdates.send(update);
```

Comment:
Will this update also be received on the subscription below, duplicating the update on `updatesController`, or is there something preventing this?

The simplest solution I can think of is just removing the `updatesController.add(update)`. This may add a small amount of latency for the update in the current tab, but prevent the duplicates.

Reply:
During testing, it looked like only other tabs receive messages sent through the broadcast channel (I had to add the `updatesController.add` line to get local updates to work).

That's not supposed to be the case though, I'll figure out what happened there and how to fix this properly.

Reply:
Nevermind, [this page](https://developer.chrome.com/blog/broadcastchannel) says: 

>A channel won't broadcast to itself. So if you have an onmessage listener on the same page as a postMessage() to the same channel, that message event doesn't fire.

That's also the behavior I'm seeing when pasting this into the console:

```JS
const channel = new BroadcastChannel('test');
channel.onmessage = console.log;
channel.postMessage('test');
```

I've added a comment to explain that we really have to listen on both streams.

Reply:
Great, thanks for confirming!